sourcefiles/ctoptions.py

```diff
@@ -0,0 +1,450 @@
+'''
+Module for preconfiguring in-game options at compile time
+'''
+
+from ctenums import ActionMap, InputMap
+from ctrom import CTRom
+
+'''
+TODO fill in explaination of controller remapping CT uses
+'''
+
+class ControllerBinds:
+    def __init__(self, data: bytearray = None):
+
+        self._mappings = self.get_vanilla()
+
+        '''
+        #only support a single parameter, but optionally
+        if args:
+           data = args[0]
+        '''
+        
+        if data is None:
+            return
+    
+        #only support 8 or 11 bytes
+        if len(data) == 11:
+            self.update_from_bytes(data[3:])
+        elif len(data) == 8:
+            self.update_from_bytes(data[:])
+        else:
+            raise ValueError('Button mappings must be either 8 or 11 bytes in length')
+            exit()
+
+    @classmethod
```

Comment:
Going to change this to @staticmethod since it's not using any class variables.

Reply:
Oh, that exists. Cool. Changed.